tests/server-integration/Utils.hs

```diff
@@ -0,0 +1,121 @@
+  ) where
+
+import Data.Proxy
+import Data.Text.IO qualified as TIO
+import Network.Wai.Handler.Warp
+import Servant.Server
+
+import Backend.Commands
+import Config
+import Control.Lens
+import Control.Monad (void)
+import Data.Aeson (Value)
+import Data.Aeson.Lens
+import Data.Maybe (fromMaybe)
+import Data.Text (Text)
+import Data.Text qualified as T
+import GHC.IO.Handle (Handle)
+import Network.HTTP.Req
+import System.Environment (lookupEnv)
+import System.Exit
+import System.Process
+  (CreateProcess(std_err, std_out), ProcessHandle, StdStream(NoStream), callProcess, createProcess,
+  getPid, proc)
+import Test.Tasty.HUnit (assertFailure)
+import Toml qualified
+import Web.API
+import Web.Server
+
+readConfig :: FilePath -> IO Config
+readConfig configPath = do
+  text <- TIO.readFile configPath
+  let config = Toml.decode configCodec text
+  case config of
+    Left err -> do
+      TIO.putStrLn $ "An error occurred while decoding '" <> T.pack configPath <> "':"
+      TIO.putStrLn $ Toml.prettyTomlDecodeErrors err
+      exitFailure
+    Right config -> pure config
+
+getConfigPath :: IO FilePath
+getConfigPath = do
+  envConfigPathMb <- lookupEnv "COFFER_CONFIG"
+  pure $ fromMaybe "config.toml" envConfigPathMb
+
+runServer :: IO ()
+runServer = do
+  configPath <- getConfigPath
+  -- TODO: try sending config via api.
+  config <- readConfig configPath
+  run 8081 do
+    serve (Proxy :: Proxy API) do
+      makeServer \_token -> reportErrors . runBackendIO' . runCommand config
+
+procWithoutOutput :: CreateProcess -> CreateProcess
+procWithoutOutput cp = cp
+  { std_out = NoStream
+  , std_err = NoStream
+  }
+
+-- TODO: figure out why we can't access vault
+runVault :: Int -> String -> IO (Maybe Handle, Maybe Handle, Maybe Handle, ProcessHandle)
+runVault port tokenId = createProcess process
+  where
+    process = procWithoutOutput $
+      proc "vault"
+        [ "server"
+        , "-dev"
+        , "-dev-root-token-id=" <> tokenId
+        , "-dev-listen-address=localhost:" <> show port
+        ]
+
+killVault :: ProcessHandle -> IO ()
+killVault handle = do
+  mPid <- getPid handle
+  case mPid of
+    Nothing -> pure ()
+    Just pid -> do
+      callProcess "kill" [show pid]
+
+deleteRecords :: IO ()
+deleteRecords = runReq defaultHttpConfig do
+  void
+    $ req
+      DELETE
+      (http "localhost" /: "api" /: "v1" /: "content" /: "delete")
+      NoReqBody
+      ignoreResponse
+      ( mconcat
+          [ header "token" "root"
+          , "path" =: ("/" :: Text)
+          , queryFlag "recursive"
+          , port 8081
+          ]
+      )
+
+cleanEntry :: Value -> Value
+cleanEntry value =
+  value
+    & key "eDateModified" .~ ""
+    & key "eFields" . _Object . each . key "fDateModified" .~ ""
+
+getValueContents :: Value -> IO Value
+getValueContents value = do
+  case value ^? key "contents" of
+    Just res -> pure res
+    Nothing -> assertFailure $ "Expected field \"contents\", but don't get it: " <> show value
```

Comment:
I don't think we need this function. I would just add the "contents" key (and the "tag" key) to the expected json in the `unit_view_an_entry` test, in `expected = [aesonQQ| { "contents": ...`

Reply:
Or don't, I guess it doens't matter much, this is going away in #110 anyway, so it's ok.